### PR TITLE
Add metrics nodes support

### PR DESCRIPTION
**Pull request by WillianFuks**
This is a proposal for solving issue #104 . The new code considers metric resources and adds them to the dependencies of the tested node which makes possible to test models that relies on metrics now.

Integration tests have already been implemented as well. Please feel free to ask for changing anything.

Best,

### CODE REVIEW

**Comment by cdiniz** [comment]
Hi @WillianFuks,
Thank you so much for your effort to add metrics support to the library. 
We'll review the code and get back to you soon.

(Sorry about the delay)

File: integration-tests/ci/profiles.yml [hunk 1]
```diff
@@ -7,18 +7,18 @@ integration_tests:
       project: "{{ env_var('GOOGLE_CLOUD_PROJECT') }}"
       dataset: dbt_unit_testing
       threads: 1
-    
+
     postgres:
       type: postgres
-      host: localhost
+      host: postgres
```
Comment (cdiniz) [inline integration-tests/ci/profiles.yml:13]: Could you please revert this to localhost? It's needed for the CI env.

Comment (WillianFuks) [reply integration-tests/ci/profiles.yml]: Just now I saw this message. Already pushed the required change.

**Comment by cdiniz** [comment]
Hi @WillianFuks, I've updated the repo to run tests with updated versions of dbt and dbt-[providers]. I've also updated the way we were running tests in different versions.
Currently, the main issue I'm finding is the following: 
- Metrics tests are not running, the test selector for metrics is not finding any tests. I suspect it's something about moving metrics folders from test fixtures to models.
- There's an error about dbt/metrics compatible versions.

If you can take a look, and consider removing the 'folder' copy (which was required in past), we appreciate it and we continue to review the pr!

Thank you for your support.

**Comment by cdiniz** [comment]
Thank you for your collaboration, we're closing this due to inactivity.

**Pull request closed without merging**